File: dev/Basic/shared/entities/androidCommunicator/communicator/message/base/Handler.cpp
/*
 * Handler.cpp
 *
 *  Created on: May 9, 2013
 *      Author: vahid
 */
#include "Handler.hpp"
#include "Message.hpp"
namespace sim_mob
{

Handler::Handler(sim_mob::comm::Message* message_):message(message_){
}

sim_mob::comm::Message * Handler::getMessage()
{
	return message;
}

}

File: testing/SimMobility4Android/simmobility4android-lib/src/main/java/sg/smart/mit/simmobility4android/handler/Handler.java
package sg.smart.mit.simmobility4android.handler;

import sg.smart.mit.simmobility4android.connector.Connector;
import sg.smart.mit.simmobility4android.message.Message;

/**
 * Interface for all handler implementations
 *
 * @author gandola
 */
public abstract class Handler<T extends Message> {

    private T message;
    private Connector connector;

    public Handler(T message, Connector connector) {
        this.message = message;
        this.connector = connector;
    }

    public abstract void handle();

    protected Connector getConnector() {
        return connector;
    }

    public T getMessage() {
        return message;
    }
}
